### ISSUE TITLE
`&lt;` will be transformed to `<` in linkify-html

### ISSUE THREAD

**Issue by evshiron** [Hypercontext/linkifyjs#371]
1. The tokenizer you use will parse HTML entities and emit them as `Chars` token.

Try it with [RunKit](https://npm.runkit.com/%40nfrasser%2Fsimple-html-tokenizer) and the following code:
```javascript
console.log(simpleHtmlTokenizer.tokenize('&lt;div /&gt;'));
```
The result will be a `Chars` token of `<div />`.

2. `linkify-html` doesn't escape `Chars` tokens and output them as is, as shown below:

https://github.com/Hypercontext/linkifyjs/blob/main/packages/linkifyjs/src/linkify-html.js#L190

3. Thus enable XSS for previously escaped HTML string, which is bad.

```javascript
linkifyHtml('<div>&lt;script&gt;console.log("🦄")&lt;/script&gt;</div>');
```

**Comment by nfrasser** [comment]
@evshiron thanks for reporting, will keep an eye on this but this _should_ be okay right now. 

`linkifyjs` v3 depends on `@nfrasser/simple-html-tokenizer` v0.5.11-1 (strictly) which parses the characters as `'&lt;div /&gt;` and does not have this XSS problem.

The upcoming `linkifyjs` v4 (not released yet) depends on `@nfrasser/simple-html-tokenizer` v0.5.11-2, which does parse the characters as you noted, but now also converts them back to their encoded-versions before outputting the text

Relevant commit in the v4 branch: https://github.com/Hypercontext/linkifyjs/commit/eccf1278c7da83b2cb48bdd87cf35f6c89668176

See #354

**Comment by evshiron** [comment]
A minimal reproducible example:

https://codesandbox.io/s/blissful-microservice-96ovy?file=/src/index.js

**Comment by nfrasser** [comment]
@evshiron you were absolutely right, `simple-html-tokenizer` was inadvertently bumped in linkifyjs@v3.0.4. Just published v3.0.5 which reverts this change. v3.0.4 is the only affected version. Thanks for reporting and reproducing!
